resource_estimator/src/estimates/factory/dispatch.rs

```diff
@@ -0,0 +1,101 @@
+use crate::estimates::{ErrorCorrection, Factory, FactoryBuilder};
+use std::borrow::Cow;
+
+#[derive(Clone)]
+pub enum FactoryDispatch2<F1, F2> {
```

Comment:
Not sure if enum name with index and enums themselves are descriptive. Can we have maybe much longer name but self-descriptive?

Reply:
I cannot come up with a better name, so I added a doc string to make clearer what this is used for. Eventually there might be `FactoryDispatch3` and `FactoryDispatch4`, but since there is no use case for more than 2 magic state types for now, I have not added them yet.

Reply:
I see. Thank you!
 A couple ideas:
1.  Would it be easier to read if generic parameters would be Factory1 and Factory2 instead of F1 and F2?
2. Same for enum

Or FactoryType1 and FactoryType2

Reply:
Thanks, I updated the code and use the longer names for the generic parameters, both for the factory and the builder.